Various Doc Enhancements (#132)
Azure/Azure-Verified-Modules: docs/content/contributing/q-and-a.md

---

### What do I need so I can start developing a module?

We suggest that you review [module specification][ModuleSpecifications] and [contribution guides][ModuleContributors]:

- [Bicep contribution guide][BicepContributios]
- [Terraform contribution guide][TerrafromContribution]

Feel free to reach out to the [AVM Core team][AVMCoreTeam] in case that additional help is needed.

---

### What do I do about existing modules that are available doing a similar thing to my module that I am proposing to develop and release?

> We suggest that you first review the FAQ of [What is happening to existing initiatives like CARML and TFVM?][CARMLTFVMEvolution] to understand AVMs intent for existing modules from known Microsoft projects etc.

Reviewer: ```suggestion
We suggest that you first review the FAQ of [What is happening to existing initiatives like CARML and TFVM?][CARMLTFVMEvolution] to understand AVM's intent for existing modules from known Microsoft projects etc.
```

> As part of the [Module Proposal process][ProcessOverview], the AVM core team will work with you to triage your proposal and also to ensure there are no existing modules, from known Microsoft projects, to migrate to AVM.

Reviewer: ```suggestion
As part of the [Module Proposal process][ProcessOverview], the AVM core team will work with you to triage your proposal. We also want to make sure that no similar existing modules from known Microsoft projects are already on their way to be migrated to AVM.
```

- If there aren't any, then you can proceed with developing your module from scratch once given approval to proceed by the AVM core team.
- However, if there are existing modules from Microsoft projects we would invite you to help us complete the migration to AVM of this module; this may also entail working with the existing module owner/team.

> For existing modules that may not be directly owned and developed by Microsoft or their employees you should first review the license applied to the GitHub repository hosting the module and understand its terms and conditions. More information on GitHub repositories and licenses can be found here in [Licensing a repository][GitHubLicensing]. Most modules will use a license that will allow you to take inspiration and copy all or parts from the module source code however, to confirm, you should always check the license and any conditions you may have to meet by doing this.

Reviewer: ```suggestion
For existing modules that may not be directly owned and developed by Microsoft or their employees you should first review the license applied to the GitHub repository hosting the module and understand its terms and conditions. More information on GitHub repositories and licenses can be found here in [Licensing a repository][GitHubLicensing]. Most modules will use a license that will allow you to take inspiration and copy all or parts from the module source code. However, to confirm, you should always check the license and any conditions you may have to meet by doing this.
```

You **SHOULD NOT** use any additional labels.

> There is also a [PowerShell script][MandatoryLabels] that the [AVM core team][AVMCoreTeam] created that can help to apply those labels to module GiHub repository.

Reviewer: ```suggestion
There is also a [PowerShell script][MandatoryLabels] that the [AVM core team][AVMCoreTeam] created that can help to apply those labels the GitHub module repository.
```
